src/sparseml/recipe_template/cli.py

```diff
@@ -12,15 +14,61 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import logging
+
+import click
 from sparseml import recipe_template
+from sparseml.version import __version__
+
+
+_LOGGER = logging.getLogger(__name__)
 
 
-def main(*args, **kwargs):
+@click.command(context_settings=(dict(show_default=True)))
+@click.version_option(version=__version__)
+@click.option(
+    "--pruning",
+    # other algos will be added in future
+    type=click.Choice(["true", "false", "gmp"], case_sensitive=False),
+    default="false",
+    help="Specify if recipe should include pruning steps, can also take in the "
+    "name of a pruning algorithm",
+)
+@click.option(
+    "--quantization",
+    type=click.Choice(["true", "false"], case_sensitive=False),
+    default="false",
+    help="Specify if recipe should include quantization steps, can also take in "
+    "the name of the target hardware",
+)
```

Comment:
Use `is_flag=True, default=False, type=bool`

Reply:
Good catch, thanks!

Reply:
This is a requirement in the acceptance doc, Corey, will bring it up and make the change if approved